cpp/src/sampling/detail/update_temporal_edge_mask_impl.cuh

```diff
@@ -0,0 +1,97 @@
+/*
+ * Copyright (c) 2025, NVIDIA CORPORATION.
+ *
+ * Licensed under the Apache License, Version 2.0 (the "License");
+ * you may not use this file except in compliance with the License.
+ * You may obtain a copy of the License at
+ *
+ *     http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS IS" BASIS,
+ * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ * See the License for the specific language governing permissions and
+ * limitations under the License.
+ */
+
+#pragma once
+
+#include "prims/transform_e.cuh"
+#include "prims/update_edge_src_dst_property.cuh"
+
+#include <cugraph/detail/utility_wrappers.hpp>
+#include <cugraph/utilities/device_functors.cuh>
+#include <cugraph/utilities/mask_utils.cuh>
+
+#include <raft/core/handle.hpp>
+
+#include <thrust/scatter.h>
+
+#include <limits>
+
+namespace cugraph {
+namespace detail {
+
+template <typename vertex_t, typename edge_t, typename edge_time_t, bool multi_gpu>
+void update_temporal_edge_mask(
+  raft::handle_t const& handle,
+  graph_view_t<vertex_t, edge_t, false, multi_gpu> const& graph_view,
+  edge_property_view_t<edge_t, edge_time_t const*> edge_start_time_view,
+  raft::device_span<vertex_t const> vertices,
+  raft::device_span<edge_time_t const> vertex_times,
+  edge_property_view_t<edge_t, uint32_t*, bool> edge_time_mask_view)
+{
+  edge_time_t const STARTING_TIME{std::numeric_limits<edge_time_t>::min()};
+
+  edge_src_property_t<edge_t, edge_time_t, false> edge_src_times(handle, graph_view);
+
+#if 0
+  // FIXME:  This call to update_edge_src_property seems like what I want, but it
+  //         doesn't work.
```

Comment:
Do you know why? What happened with this?

Reply:
This variation of the call is only used in a few places.  MG SSSP, and Random Walks are the only places I see it.

Not sure if it's broken there and we haven't noticed, or if I'm doing something wrong here.  But I printed the relevant contents of the arrays and they were incorrect.  Since I use them as indices into device arrays, I could clearly see that the wrong values affected the output.

I spent a little time trying to diagnose, but didn't have time to really dig in.  It's possible I've done something incorrect in the setup, but setting do_expensive_check didn't reveal anything wrong.

Reply:
How can I reproduce this? I can dig in to see what's going awry here.

Reply:
I'll try and create a reproducer for you.

Reply:
We should debug this, the else path is clearly more expensive.

And in case we have both edge start & end time, can we just pass both the start & end time as edge property values and check (seed_time > edge start time) && (seed time < edge end time)?

Then, we don't need to perform temporal_partition_vertices nor update_temporal_edge_mask; much simpler code. Or aren't we doing this as this doubles the cases we need to consider (so double the binary size in temporal sampling edges?)

Reply:
Or we may set the mask based on the minimum time attached with a seed vertex (so we can reduce the size of the extracted subgraph in the sampling primitive).